test/listen.1.test.js

```diff
@@ -89,3 +89,29 @@ test('listen after Promise.resolve()', t => {
       })
     })
 })
+
+test('listen works with undefined host', async t => {
+  process.on('warning', () => {
+    t.fail('should not be deprecated')
+  })
+
+  const fastify = Fastify()
+  t.teardown(fastify.close.bind(fastify))
+  await fastify.listen({ host: undefined, port: 0 })
+  const address = fastify.server.address()
+  t.equal(address.address, localhost)
+  t.ok(address.port > 0)
+})
+
+test('listen works with null host', async t => {
+  process.on('warning', () => {
+    t.fail('should not be deprecated')
+  })
+
+  const fastify = Fastify()
+  t.teardown(fastify.close.bind(fastify))
```

Comment:
on teardown, we should remove the `process` listener too on both tests

Reply:
@Eomm I don't get it, it's the same as we do in all`listen.*.test.js` tests

Reply:
Those tests are wrong too, here is what I meant:

```js
test('listen works with undefined host', async t => {
  const doNotWarn = () => {
    t.fail('should not be deprecated')
  }
  process.on('warning', doNotWarn)

  const fastify = Fastify()
  t.teardown(() => {
    fastify.close()
    process.removeListener('warning', doNotWarn)
  })
  await fastify.listen({ host: undefined, port: 0 })
  const address = fastify.server.address()
  t.equal(address.address, localhost)
  t.ok(address.port > 0)
})
```

If a warning happens in another test case, the handler of this one would also be triggered, causing fuzzy behaviours.

Reply:
Uhm... Clear, but other concurrent tests could trigger anyway in the lapse of time between adding and removing the listener, am I wrong?

Reply:
> Clear, but other concurrent tests could trigger anyway in the lapse of time between adding and removing the listener

No, because each test file runs in its own nodejs child process and has its own `process` instance (this behaviour is granted by `tap`). Then the `test()` functions in each file run serially, so the concurrency is not an issue in our test suite 👍🏼 

> Maybe listening for warnings on the process is not a good way (?).

It is the right way TBH, the other options would be to mock or monkey patch the `process` itself but we avoid it.

Reply:
All clear, thanks for the explanation! I'll change the 2 tests of this PR and create a new PR for fixing other tests that don't remove the warning listener.